LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -56,71 +56,72 @@ public void CreateAccounts(int accountCount)
-            Console.WriteLine("********  Finished creating accounts   ********\n");
+            ReportSummary("Account", accountCount);
+            ReportFooter("Create Account");
         }
 
         public void ActivateAccounts()
         {
-            Console.WriteLine("******** Started activating accounts ********\n");
+            ReportHeader("Activate account");
             int pendingAccountQueueCount = pendingAccountQueue.Count;
+            int i = 0;
             while(pendingAccountQueue.Count > 0)
             {
-                Console.WriteLine("========== Activate Account =======");
                 Account account = pendingAccountQueue.Dequeue();
                 account.Status = Status.Active;
-                Console.WriteLine("\n========== Activated account =======\n{0}\n", account);
+                ReportItem(account.ToString(), "Activated account", i++);
             }
-            Console.WriteLine("========== Activated {0} accounts  =======\n", pendingAccountQueueCount);
-            Console.WriteLine("******** Finished activating accounts ********\n");
+            ReportFooter("Activate account");
         }
 
         public void CreateSubjects(int subjectCount)
         {
-            Console.WriteLine("******** Started creating subjects  ********\n");
+            ReportHeader("Subject generation");
             int i = 0;
             while (i < subjectCount)
             {
                 Subject subject = ObjectGenerator.GenerateSubject();
-                Console.WriteLine("========== Created  Subject {0} =======\n\n{1}\n", i, subject);
+                ReportItem(subject.ToString(), "Subject", i++);
                 subjectList.Add(subject);
                 subjectIdList.Add(subject.Id);
                 i++;
             }
-            Console.WriteLine("========== Generated {0} Subjects  =======\n", subjectCount);
-            Console.WriteLine("******** Finished creating subjects  ********\n");
+            ReportSummary("Subject", subjectCount);
+            ReportFooter("Subject generation");
         }
 
         public void CreateModules(int moduleCount)
         {
-            Console.WriteLine("******** Started creating modules  ********\n");
-            int i = 0;
-            while (i < moduleCount)
+            if(subjectIdList.Count != 0)
             {
-                Guid subjectId = subjectIdList[AttributeGenerator.random.Next(0, subjectIdList.Count)];
-                Module module = ObjectGenerator.GenerateModule(subjectId);
-                moduleDictionary.Add(module.Id, module);
-                moduleIdList.Add(module.Id);
-                Console.WriteLine("\n========== Created  Module {0} =======\n\n{1}\n", i, module);
-                i++;
-            }
-            Console.WriteLine("========== Generated {0} Modules  =======\n", moduleCount);
-            Console.WriteLine("******** Finished creating modules ********\n");
+                ReportHeader("Module generation");
```

Comment:
You should always report the header, footer and summary, even when your validation fails and you can't create modules. Therefore the order should be:
1. ReportHeader();
2. Do validation:
 if(subjectIdList.Count != 0) 
{
     ReportItem();
} 
else
{
     ReportError();
}
ReportSummary();
ReportFooter();